java-generator/it/src/it/preserve-unknown-objects/src/test/java/io/fabric8/it/certmanager/TestPreserveUnkonwnUsability.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.it.certmanager;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.dapr.v1alpha1.Component;
+import io.dapr.v1alpha1.ComponentSpec;
+import io.dapr.v1alpha1.componentspec.Metadata;
+import io.fabric8.kubernetes.api.model.AnyType;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.Test;
+import io.fabric8.java.generator.testing.KubernetesResourceDiff;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class TestPreserveUnknownUsability {
+
+  @Test
+  void testDeserialization() {
+    // Arrange
+    Component sample =
+      Serialization.unmarshal(getClass().getResourceAsStream("/sample.yaml"), Component.class);
+
+    // Act
+    List<Metadata> metadataList = sample.getSpec().getMetadata();
+
+    // Assert
+    assertEquals(5, metadataList.size());
+  }
+
+  private Component createSampleComponent() throws Exception {
```

Comment:
@shawkins this test also validates your good work! Thanks!

Reply:
@andreaTP thanks for picking this up